### DISCUSSION TITLE
Search for articles with no tags

### DISCUSSION THREAD

**Discussion by njnygaard** [zk-org/zk#449] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

It would be very useful to me to search for articles without tags.
I have articles with tags of the form:

```yaml
tags: some list of tags
```

and

```yaml
tags:
  - that
  - have
  - more
  - structure
```

I can probably take on this feature, I just want to know if there are other ideas before I dive in with a PR. Ideally, this functionality exists and I don't know how to access it.

```bash
zk list -f oneline -q -t "" | wc -l
```

This just gives me all the articles I have.

**Comment by tjex** [comment]
Cool. I can see some use cases for this, and think it would be a valuable addition to the feature set.

I'm not sure there's already a way to filter directly for this. Also, I can't run the command as you've shown, I get an error 🧐:

```text
base > zk list -f oneline -q -t ""
zk: error: --tag: unexpected EOL
```

I would think you could use the implementation of the `zk list --orphan` flag as an inspiration / guide as how to approach this. As it's also filtering based on an absence of incoming links. So perhaps the implementation can show how to approach filtering on the absence of tags.

I would think this should be accessed via a flag, like `zk list --no-tags` / `zk list -nt`.

**Comment by njnygaard** [comment]
Cool @tjex, I'll cut some time out of my weekends.
I don't know about the `EOL`, still working on mine:
![image](https://github.com/user-attachments/assets/fb052d55-655b-434f-8a1a-2457c2effc63)
![image](https://github.com/user-attachments/assets/2ee5b6a2-65ca-42af-9fda-96d82c1b2e61)

**Comment by njnygaard** [comment]
https://github.com/zk-org/zk/pull/450

**Comment by njnygaard** [comment]
@tjex working locally for me. I'm not sure who to tag for the review.

**Reply by tjex** [reply]
#450 is merged :)
